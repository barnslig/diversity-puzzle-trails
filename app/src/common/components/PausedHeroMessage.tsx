import { Box, Button } from "@mui/material";
import { FormattedMessage } from "react-intl";
import * as React from "react";

import ButtonProgressIndicator from "./ButtonProgressIndicator";
import HeroMessage from "./HeroMessage";
import useUpdateClock from "../hooks/api/useUpdateClock";

const PausedHeroMessage = () => {
  const [isLoading, updateClock] = useUpdateClock();

  return (
    <HeroMessage
      title={
        <FormattedMessage
          defaultMessage="Spiel pausiert"
          description="game paused message title"
        />
      }
      description={
        <FormattedMessage
          defaultMessage="Das Spiel ist pausiert. Du kannst es aber für alle fortsetzen."
          description="game paused message description"
        />
      }
      after={
        <Box marginTop={4} marginBottom={6}>
          <Button
            variant="outlined"
            color="primary"
            onClick={() => updateClock("running")}
            disabled={isLoading}
          >
            <FormattedMessage
              defaultMessage="Spiel für alle fortsetzen"
              description="game paused message continue button label"
            />
            {isLoading && <ButtonProgressIndicator />}
          </Button>
        </Box>
      }
    />
  );
};

export default PausedHeroMessage;
