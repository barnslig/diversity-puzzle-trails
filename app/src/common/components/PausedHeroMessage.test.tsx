import * as React from "react";
import render from "../testing/render";

import PausedHeroMessage from "./PausedHeroMessage";

it("renders correctly", () => {
  const { container } = render(<PausedHeroMessage />);
  expect(container.firstChild).toMatchSnapshot();
});
