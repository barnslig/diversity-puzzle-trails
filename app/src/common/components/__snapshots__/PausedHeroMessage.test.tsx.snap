// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="MuiBox-root css-elj48x"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
    >
      Spiel pausiert
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Das Spiel ist pausiert. Du kannst es aber für alle fortsetzen.
    </p>
    <div
      class="MuiBox-root css-fv0itm"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Spiel für alle fortsetzen
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</main>
`;
